packages/react/src/components/ContainedList/ContainedList.tsx

```diff
@@ -5,14 +5,24 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import React from 'react';
+import React, { ReactNode } from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
 import { LayoutConstraint } from '../Layout';
 import { useId } from '../../internal/useId';
 import { usePrefix } from '../../internal/usePrefix';
 
-const variants = ['on-page', 'disclosed'];
+const variants: ('on-page' | 'disclosed')[] = ['on-page', 'disclosed'];
+
+interface ContainedListProps {
+  action?: ReactNode;
+  children?: ReactNode;
+  className?: string;
+  isInset?: boolean;
+  kind?: 'on-page' | 'disclosed';
+  label: ReactNode;
+  size?: 'sm' | 'md' | 'lg' | 'xl';
+}
```

Comment:
Could you copy the comments from the proptypes? It's what will get shown in intellisense.

Reply:
Sure, will do it.